backend/server/src/services/tournamentService.ts

```diff
@@ -1,13 +1,35 @@
 import NotFoundError from "../errors/NotFoundError.js";
-import { type CreateTournamentRequest } from "../models/requestModel.js";
-import { TournamentModel, type Tournament } from "../models/tournamentModel.js";
-import UserModel from "../models/userModel.js";
+import {
+  TournamentModel,
+  type Tournament,
+  type UnsavedMatch,
+  TournamentType
+} from "../models/tournamentModel.js";
+import UserModel, { type User } from "../models/userModel.js";
 import BadRequestError from "../errors/BadRequestError.js";
+import { type Types } from "mongoose";
+import MatchModel from "../models/matchModel.js";
+import { type CreateTournamentRequest } from "../models/requestModel.js";
+import { type Match, type MatchPlayer } from "../models/matchModel.js";
 
 export class TournamentService {
   public async getTournamentById(id: string): Promise<Tournament> {
-    const tournament = await TournamentModel.findById(id).exec();
-
+    const tournament = await TournamentModel.findById(id)
+      .populate<{ creator: User }>({ path: "creator", model: "User" })
+      .populate<{ players: User[] }>({ path: "players", model: "User" })
+      .populate<{
+        matchSchedule: Match[];
+        "matchSchedule.players": User[];
+        "matchSchedule.winner": User;
+      }>({
+        path: "matchSchedule",
+        model: "Match",
+        populate: [
+          { path: "players", model: "User" },
+          { path: "winner", model: "User" }
+        ]
+      })
+      .exec();
```

Comment:
This works now which is great, but we can remove the winner and player populations. The player population would require some changes to the match schema for it to work. However, we can just infer the player info from the tournaments player collection. Same thing goes for the winner info.